src/main/java/com/ibm/disni/rdma/verbs/impl/RdmaCmNat.java

```diff
@@ -106,7 +99,12 @@ public IbvQP createQP(RdmaCmId id, IbvPd pd, IbvQPInitAttr attr)
 		if (!natRecvCq.isOpen()) {
 			throw new IOException("Trying to create a QP with closed receive CQ");
 		}
-		long objId = nativeDispatcher._createQP(idPriv.getObjId(), natPd.getObjId(), natSendCq.getObjId(), natRecvCq.getObjId(), attr.getQp_type(), attr.cap().getMax_send_wr(), attr.cap().getMax_recv_wr(), attr.cap().getMax_inline_data());
+		IbvQPCap cap = attr.cap();
+		long objId = nativeDispatcher._createQP(idPriv.getObjId(), natPd.getObjId(), natSendCq.getObjId(),
+				natRecvCq.getObjId(), attr.getQp_type(), cap.getMax_send_wr(), cap.getMax_recv_wr(), cap.getMax_inline_data(), cap.getMax_send_sge());
+		if (objId == -1) {
+			throw new IOException("Failed to create a QP");
+		}
```

Comment:
While throwing an exception at this point (rather than just returning null) is probably the right decision, I'd prefer to remove this aspect from this PR and then later have a separate PR that consistently looks at exceptions throughout RdmaCmNat and RdmaVerbsNat because there are many calls that could be migrated to an exception model. Let's think if it makes sense to broadly move towards an exception model in those two classes and treat this in a seperate PR.